### PR TITLE
Factotum migrate to nwfirebase

### PR DESCRIPTION
**Pull request by byronwang93**
# TL;DR
- removed usage of mongoDB in factotum -> replaced it to point to current nwFirebase (mainly used in initializing a discord bot in a new server)
- wrote `firebaseUtil.js` which replaces the `mongoUtil.js` functionality translated to firebase functions
- removed all references in codebase of mongo files (`BotGuild.d.ts`, `BotGuild.js` and `mongoUtil.js`)
- removed all references in codebase of old firebase (this was firebase-services.js) + moved the functions to firebaseUtil.js

# Description
New Feature/Bug Fix ...

## Things NOT fixed / broken
- in App.js, I commented out await botGuild.setCommandStatus(bot);
- mongo files references have been removed but the files still exist (might want to delete)
- functions from firebase-services.js have been migrated to firebaseUtil.js (might want to delete this file)

## Tests Done

# Breaking Changes
- Removing MongoDB code includes removing usage of `mongoose`, a library for defining schemas. We might want to consider looking into a firebase alternative in the future

- Changed method for starting program - mode config (dev/test/prod + test servers) are now configured in env file instead of as command line parameters

### CODE REVIEW

File: db/firebase/firebaseUtil.js [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+        return Promise.resolve();
+    },
+
+    // FIREBASE SERVICE FUNCTIONS IMPORTED BELOW
+    /**
+     * Retrieves a question from the db that has not already been asked at the Discord Contests, then marks the question as having been 
+     * asked in the db.
+     * @param {String} guildId - the id of the guild
+     * @returns {Object | null} - the data object of a question or null if no more questions
+     */
```
Comment (byronwang93) [inline db/firebase/firebaseUtil.js:113]: the rest of the functions here have been copy + pasted from firebase-services.js file except GetBotGuild + CreateBotGuild

**Comment by martincai8** [comment]
you're so good at coding

File: db/firebase/firebaseUtil.js [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+    async retrieveLeaderboard(guildId) {
+        const snapshot = (await getFactotumDoc().collection('guilds').doc(guildId).collection('questionsLeaderboard').get()).docs;
+        const winners = snapshot.map(doc => doc.data()).sort((a, b) => b.points - a.points);
+        return winners;
+    },
+
+};
+
+function getFactotumDoc() {
+    return apps.get('nwPlusBotAdmin').firestore().collection('ExternalProjects').doc('Factotum');
```
Comment (daniel-panhead) [inline db/firebase/firebaseUtil.js:372]: Should we use the _db object or directly reference with apps.get?

**Review by daniel-panhead — APPROVED** [review]
Great coder

File: db/firebase/firebaseUtil.js (hunk 2 above)
Comment (byronwang93) [reply db/firebase/firebaseUtil.js]: yep just addressed @daniel-panhead ^^
